core/state/state_object.go

```diff
@@ -264,15 +264,15 @@ func (s *stateObject) finalise(prefetch bool) {
 	}
 }
 
-// updateTrie writes cached storage modifications into the object's storage trie.
-// It will return nil if the trie has not been loaded and no changes have been
-// made. An error will be returned if the trie can't be loaded/updated correctly.
-func (s *stateObject) updateTrie() (Trie, error) {
-	// Make sure all dirty slots are finalized into the pending storage area
-	s.finalise(false) // Don't prefetch anymore, pull directly if need be
```

Comment:
This check that you are removing, I think it is fine to do so, since we should have called finalize once per transaction. which should have fixed this. 
However, we may not have called finalize after the consensus-engine does it's thing. Example ethash adds a reward to an account, and we haven't called `Finalise` after that. That is probably fine, though, since doling out miner rewards does not touch storage tries, but I just wanted to raise it anyway, so we don't forget that state modifications happen there too. 

Might be a good idea to add a panic here in case the call to `s.finalise(false)` is _not_ a no-op, and then run all the tests and see if something ever triggers it. 
In fact, I'll do that locally.

Reply:
```

Reply:
Thank you for pointing it out. I will double check it.

Reply:
I added this check, and there is no panic in my branch by running `go test ./tests/...`

But I think the issue you raised is valid, it's possible we might modify the storage in the post-transaction stage. I will just add it back.